src/commonMain/kotlin/kotools/types/range/NotEmptyRange.kt

```diff
@@ -0,0 +1,59 @@
+package kotools.types.range
+
+import kotools.types.SinceKotoolsTypes
+
+/**
+ * Representation of a range of comparable values that contain at least one
+ * value.
+ */
+@SinceKotoolsTypes("4.2")
+public data class NotEmptyRange<T : Comparable<T>> internal constructor(
```

Comment:
Again, we should convert the `NotEmptyRange` type from a data class to a sealed interface.